base/errorshow.jl

```diff
@@ -54,19 +54,19 @@ end
 function showerror(io::IO, ex::TypeError)
     print(io, "TypeError: ")
     if ex.expected === Bool
-        print(io, "non-boolean ($(typeof(ex.got))) used in boolean context")
+        print(io, "non-boolean (", typeof(ex.got), ") used in boolean context")
     else
         if isa(ex.got, Type)
-            tstr = "Type{$(ex.got)}"
+            targs = ("Type{", ex.got, "}")
         else
-            tstr = string(typeof(ex.got))
+            targs = (typeof(ex.got),)
         end
         if isempty(ex.context)
-            ctx = "in $(ex.func)"
+            ctx = ("in ", ex.func)
         else
-            ctx = "in $(ex.func), in $(ex.context)"
+            ctx = ("in ", ex.func, ", in ", ex.context)
         end
```

Comment:
`.func` and `.context` fields are strings, they shouldn't have any special printing